gui/src/loader.rs

```diff
@@ -125,12 +143,30 @@ impl Loader {
         Command::none()
     }
 
-    fn on_start(&mut self, res: Result<Arc<dyn Daemon + Sync + Send>, Error>) -> Command<Message> {
+    fn on_log(&mut self, log: Option<String>) -> Command<Message> {
+        if let Step::Syncing { bitcoind_logs, .. } = &mut self.step {
+            if let Some(l) = log {
+                *bitcoind_logs = l;
+            }
+        }
+        Command::none()
+    }
+
+    fn on_start(
+        &mut self,
+        res: Result<(Arc<dyn Daemon + Sync + Send>, Option<Bitcoind>), Error>,
+    ) -> Command<Message> {
         match res {
-            Ok(daemon) => {
+            Ok((daemon, bitcoind)) => {
+                // bitcoind may have been already started and given to the loader
+                // We should not override with None the loader bitcoind field
+                if let Some(bitcoind) = bitcoind {
+                    self.internal_bitcoind = Some(bitcoind);
+                }
```

Comment:
I'm not sure i can understand this comment. Shouldn't this be checking `self.internal_bitcoind` is `None` before overriding it, if the meaning of the comment is what i think it is?

Reply:
The iced command that start_bitcoind_and_daemon, returns the daemon interface and an Option< Bitcoind> wrapper if it has started bitcoind, but If bitcoind is already started and stored in loader, the the result must be None and should not override the self.internal_bitcoind